taca/deliver/deliver.py

```diff
@@ -0,0 +1,385 @@
+                "{}_{}".format(self.sampleid,
+                    datetime.datetime.now().strftime("%Y%m%dT%H%M%S"))))
+                
+    @memoized
+    def expand_path(self,path):
+        """ Will expand a path by replacing placeholders with correspondingly 
+            named attributes belonging to this Deliverer instance. Placeholders
+            are specified according to the pattern '_[A-Z]_' and the 
+            corresponding attribute that will replace the placeholder should be
+            identically named but with all lowercase letters.
+            
+            For example, "this/is/a/path/to/_PROJECTID_/and/_SAMPLEID_" will
+            expand by substituting _PROJECTID_ with self.projectid and 
+            _SAMPLEID_ with self.sampleid
+            
+            If the supplied path does not contain any placeholders or is None,
+            it will be returned unchanged.
+            
+            :params string path: the path to expand
+            :returns: the supplied path will all placeholders substituted with
+                the corresponding instance attributes
+            :raises DelivererError: if a corresponding attribute for a 
+                placeholder could not be found
+        """
+        try:
+            m = re.search(r'(_[A-Z]+_)',path)
+        except TypeError:
+            return path
+        else:
+            if m is None:
+                return path
+            try:
+                expr = m.group(0)
+                return self.expand_path(
+                    path.replace(expr,getattr(self,expr[1:-1].lower())))
+            except AttributeError as e:
+                raise DelivererError(
+                    "the path '{}' could not be expanded - reason: {}".format(
+                        path,e))
+    
+class ProjectDeliverer(Deliverer):
+    
+    def __init__(self, projectid=None, sampleid=None, **kwargs):
+        super(ProjectDeliverer,self).__init__(
+            projectid,
+            sampleid,
+            **kwargs)
+    
+    def deliver_project(self):
+        """ Deliver all samples in a project to the destination specified by 
+            deliverypath
+            
+            :returns: True if all samples were delivered successfully, False if
+                any sample was not properly delivered or ready to be delivered
+        """
+        self.log.info(
+            "Delivering {} to {}".format(str(self),self.deliverypath))
+        try:
+            sampleentries = self.project_sample_entries()
+        except DelivererDatabaseError as e:
+            self.log.error(
+                "error '{}' occurred during delivery of {}".format(
+                    str(e),str(self)))
+            raise
+        # right now, don't catch any errors since we're assuming any thrown 
+        # errors needs to be handled by manual intervention
+        status = True
+        print(", ".join([s.get('sampleid') for s in sampleentries.get('samples')]))
```

Comment:
is this print meant to be here? I just thought that it _could_ be some debugging leftover, otherwise wouldn't it be cleaner to use `logging` instead?